addons/event/models/event_mail.py

```diff
@@ -44,16 +48,27 @@ class EventTypeMail(models.Model):
         ('before_event', 'Before the event'),
         ('after_event', 'After the event')],
         string='Trigger', default="before_event", required=True)
-    template_id = fields.Many2one(
-        'mail.template', string='Email Template',
-        domain=[('model', '=', 'event.registration')], ondelete='restrict',
-        help='This field contains the template of the mail that will be automatically sent')
+    template_model_id = fields.Many2one('ir.model', string='Template Model', compute='_compute_template_model_id')
```

Comment:
As Martin is about to remove accesses on ir.model, let us already add a compute_sudo on this one so that it does not crash due to model read. This is only a technical field anyway.